### ISSUE TITLE
EOL comment on left bracket of a block shouldn't be moved to a new line

### ISSUE THREAD

**Issue by a14n** [dart-lang/dart_style#798]
```dart
if (false) { // a
  var b = 1; // b
} // c
```

is formatted to

```dart
if (false) {
  // a
  var b = 1; // b
} // c
```

**Comment by a14n** [comment]
Same behaviour on the left square bracket of a list.

**Comment by munificent** [comment]
This is working as intended. Can you show me an example where you think it makes the code look worse?

It's probably not obvious, but one reason the formatter makes changes like this is that it lets tools like code generators produce *completely* unformatted, throw it at dartfmt, and get something nice back. Moving the line comment to the beginning of the block looks good to me. Why is it better to leave it on the previous line?

**Comment by a14n** [comment]
The problem appears also in argument lists:

```diff
           actions: <Widget>[
-            IconButton( // action button
+            IconButton(
+              // action button
               icon: Icon(choices[0].icon),
               onPressed: () { _select(choices[0]); },
             ),
-            IconButton( // action button
+            IconButton(
+              // action button
               icon: Icon(choices[1].icon),
               onPressed: () { _select(choices[1]); },
             ),
-            PopupMenuButton<Choice>( // overflow menu
+            PopupMenuButton<Choice>(
+              // overflow menu
               onSelected: _select,
               itemBuilder: (BuildContext context) {

```

See [@Hixie comment](https://github.com/flutter/flutter/pull/30138/files#r271403807):

>It makes the comments look like they're talking about the `icon`, `icon`, and `onSelected` arguments, instead of being about the  `IconButton`, `IconButton`, and `PopupMenuButton<Choice>` widgets, which is what they're supposed to be about.

I agree. EOL comment should remain on the same line as its previous token (if the previous token was on the same line before formatting)

**Comment by munificent** [comment]
Ah, interesting. That's a consequence of the fact that trailing comma argument lists are formatted the same way collection literals. For collection literals, it is deliberate that it moves the comma down since we assume it applies to the contents of the collection.

**Comment by munificent** [comment]
I spent a lot of time working on how to best handle comments at the beginning of delimited sections in [the new style](https://github.com/dart-lang/dart_style/issues/1253). It's not unreasonable to want to leave them on the same line as the opening delimiter in some cases. But I went through a *lot* of code and looked at the results of leaving them there versus moving them to the next line and overall I think it looks best to move them down.

I'm going to go ahead and close this as working as intended.
